src/librustc_middle/mir/predecessors.rs

```diff
@@ -13,37 +13,33 @@ pub type Predecessors = IndexVec<BasicBlock, SmallVec<[BasicBlock; 4]>>;
 
 #[derive(Clone, Debug)]
 pub(super) struct PredecessorCache {
-    cache: Lock<Option<Lrc<Predecessors>>>,
+    cache: OnceCell<Predecessors>,
 }
 
 impl PredecessorCache {
     #[inline]
     pub(super) fn new() -> Self {
-        PredecessorCache { cache: Lock::new(None) }
+        PredecessorCache { cache: OnceCell::new() }
     }
 
     /// Invalidates the predecessor cache.
     ///
     /// Invalidating the predecessor cache requires mutating the MIR, which in turn requires a
     /// unique reference (`&mut`) to the `mir::Body`. Because of this, we can assume that all
     /// callers of `invalidate` have a unique reference to the MIR and thus to the predecessor
-    /// cache. This means we don't actually need to take a lock when `invalidate` is called.
+    /// cache. This means we never need to do synchronization when `invalidate` is called.
```

Comment:
Aside: seems like an odd thing to put in the doc comment, we may want to move it to a regular comment.